python/mlc_chat/compiler_pass/pipeline.py

```diff
@@ -75,7 +76,8 @@ def _mlc_llm_pipeline(  # pylint: disable=too-many-arguments
     def _pipeline(mod: tvm.ir.IRModule, _ctx: tvm.transform.PassContext) -> tvm.ir.IRModule:
         seq = tvm.transform.Sequential(
             [
-                # Phase 0. Add additional information for compilation
+                # Phase 0. Add additional information for compilation and remove unused Relax func
+                PruneRelaxFunc(),
```

Comment:
how about this: let's add a boolean flag `flashinfer_enabled` in `_mlc_llm_pipeline`, which comes from `OptimizationFlags`, and base on this flag we may choose to prune flashinfer-related functions

Reply:
Just updated with a new flag here. Actually I think it might be more extensible to pass the entire compilation flag object to the pipeline, given right now we have already passed flashinfer and cublas-gemm. What do you think?

Reply:
The principle I'm trying to maintain here is that we wanted every `nn.Module` to be standalone-compilation-and-debuggable, it means:

```python
SomeModule(...).jit(format="torch", pipeline=..., pipeline_args=...)
```

and it would be ideal if `pipeline_args` are package-independent, i.e. one doesn't have to import anything extra to create a pipeline. In this sense, while there are indeed abusively huge amount of parameters to tweak, I believe we don't have to aggregate them into one dataclass that requires extra import logics - as long as we document them clearly